src/libaktualizr-posix/ipuptanesecondary.cc

```diff
@@ -52,6 +52,36 @@ Uptane::SecondaryInterface::Ptr IpUptaneSecondary::create(const std::string& add
   return std::make_shared<IpUptaneSecondary>(address, port, serial, hw_id, pub_key);
 }
 
+SecondaryInterface::Ptr IpUptaneSecondary::connectAndCheck(const std::string& address, unsigned short port,
+                                                           EcuSerial serial, HardwareIdentifier hw_id,
+                                                           PublicKey pub_key) {
+  // try to connect:
+  // - if it succeeds compare with what we expect
+  // - otherwise, keep using what we know
+  try {
+    auto sec = IpUptaneSecondary::connectAndCreate(address, port);
+    if (sec != nullptr) {
+      auto s = sec->getSerial();
```

Comment:
Mismatch of serial and hardware ID looks like critical to me, I am not sure if anything will work after such error at all?

Reply:
Changed in latest version